### ISSUE TITLE
First subscriber to a multiplexed channel adds 2 subscribers

### ISSUE THREAD

**Issue by kidd** [slact/nchan#289]
Hi,
I'm aware of #216 but I doubt it's related, so I'm opening another issue:

Given this config:
```
worker_processes  1;
master_process on;

events {
    worker_connections  1024;
}

http {
    server {
        listen       8888;

        location ~ /sub/(.*)/(.*) {
            nchan_subscriber http-raw-stream;
            nchan_channel_id $1 $2;
        }
        location = /pub {
            nchan_publisher ;
            nchan_channel_id $arg_id;
        }
    }
}
```

$ curl http://dev212:8888/sub/foo/bar

(other terminal) $  curl 'http://localhost:8888/pub?id=foo' -H'Accept: text/json' -s
{"messages": 0, "requested": 12, "subscribers": 2, "last_message_id": "0:0" }

This only happens if nchan_channel_id has 2 or more channels.  

Examples:
- nchan_channel_id $1 ;  # works ok, 1 subscriber
- nchan_channel_id $1 $2 ;  # WRONG, 2 subscribers
- nchan_channel_id $1 $2 $3;  # WRONG, 2 subscribers.

C is not my thing, but I'll try to investigate a bit more codewise. there are very few places in memstore that touch subscribers. that can be a good hook.

Cheers,

**Comment by slact** [comment]
Thanks for your report. fixed in 418329d991fe11ab57c4e9b0f059c5ad98b50000.

**Comment by kidd** [comment]
YAY! Thanks!
